components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx

```diff
@@ -435,11 +436,14 @@ export const PortfolioAsset = (props: Props) => {
   }, [nftMetadata])
 
   const currentNftPinningStatus = React.useMemo(() => {
+    if (!isAutoPinEnabled) {
+      return undefined
+    }
     if (isNftAsset && selectedAsset && nftPinnable) {
       return getNftPinningStatus(selectedAsset)
     }
     return undefined
-  }, [nftPinnable, isNftAsset, selectedAsset, nftPinningStatus])
+  }, [nftPinnable, isNftAsset, selectedAsset, nftPinningStatus, isAutoPinEnabled])
```

Comment:
nftPinningStatus is not referenced in useMemo body, need it here?

Reply:
getNftPinningStatus is dependent on nftPinningStatus i guess